FinSearchUnified/Helper/StaticHelper.php

```diff
@@ -456,199 +614,82 @@ public static function createFindologicFacet($label, $name, $type, $filter)
+    public static function cleanString($string)
     {
-        $response = [];
+        $string = str_replace('\\', '', addslashes(strip_tags($string)));
+        $string = str_replace(["\n", "\r", "\t"], ' ', $string);
 
-        foreach ($items as $subItem) {
-            $tempItem = [
-                'name' => (string) $subItem->name,
-                'image' => (string) ( $subItem->image !== null ? $subItem->image[0] : '' ),
-                'color' => (string) ( $subItem->color !== null ? $subItem->color[0] : '' ),
-                'frequency' => isset($subItem->frequency) ? (int) $subItem->frequency : null,
-                'items' => []
-            ];
-
-            if ($subItem->items->item) {
-                $tempItem['items'] = self::createFilterItems($subItem->items->item);
-            }
-
-            $response[] = $tempItem;
-        }
+        // Remove unprintable characters since they would cause an invalid XML.
+        $string = self::removeControlCharacters($string);
 
-        return $response;
+        return trim($string);
     }
 
     /**
-     * @param array $items
-     * @param string|null $name
-     * @param string|null $recurseName
+     * @param string $value
      *
-     * @return array
+     * @return null|string|string[]
      */
-    private static function prepareTreeView($items, $name, $recurseName = null)
+    public static function removeControlCharacters($value)
     {
-        $response = [];
-        $selectedItems = self::getSelectedItems($name);
-
-        foreach ( $items as $item ) {
-            $treeName = $item['name'];
-
-            if ($recurseName !== null) {
-                $treeName = $recurseName . '_' . $item['name'];
-            }
-
-            $active = in_array($treeName, $selectedItems);
-
-            if ($item['frequency'] && !$active) {
-                $label = sprintf('%s (%d)', $item['name'], $item['frequency']);
-            } else {
-                $label = $item['name'];
-            }
-
-            $treeView = new SearchBundle\FacetResult\TreeItem(
-                $treeName,
-                $label,
-                $active,
-                self::prepareTreeView($item['items'], $name, $treeName)
-            );
-            $response[] = $treeView;
-        }
-
-        return $response;
-    }
-
-    public static function checkDirectIntegration()
-    {
-        $session = Shopware()->Session();
-
-        if ($session->offsetExists('findologicDI') === false) {
-            $urlBuilder = new UrlBuilder();
-            $isDI = $urlBuilder->getConfigStatus();
-            $session->offsetSet('findologicDI', $isDI);
-            $currentIntegrationType = $isDI ? Constants::INTEGRATION_TYPE_DI : Constants::INTEGRATION_TYPE_API;
-
-            self::storeIntegrationType($currentIntegrationType);
+        if (is_string($value)) {
+            return preg_replace('/[\x00-\x1F]|[\x7F]|[\x80-\x9F]/', ' ', $value);
```

Comment:
the preg expression does not show valid in the regex tester if I use the complete range: 0xC20x80 - 0xC20x9F
what to do then?

Reply:
and I used space as a replacement because the code in the `cleanString` is using the space as a replacement. So should I change that code line as well ? 
`$string = str_replace(["\n", "\r", "\t"], ' ', $string);`

Reply:
- Yes for 1 & 2
- As for 3, there is no definition provided in the link you provided and other links I have checked that what kind of characters those are. Some simply say `"(Undefined)"`. So how would I create test cases for a character which I don't know what it is?
- I see, but it is mentioned in the youtrack task that "Control characters \n, \r and \t are stripped away." But if I use that case for the `cleanString` method, then `removeControlCharacters` will never validate that cause it is already being replaced with a space in the cleanString method.